client/src/app/app.component.ts

```diff
@@ -89,10 +74,29 @@ export class AppComponent implements OnInit {
     });
   }
 
+  private checkShowLogoutButton() {
+    return this.router.events.subscribe(event => {
+      if (event instanceof NavigationStart && (event.url === '/login' || event.url === '/logout')) {
+        this.showLogoutButton = false;
+      }
+    });
+  }
+
+  private setSsoMode() {
+    this.authenticationService.sso = true;
```

Comment:
just a thought: Would it make sense to move the whole sso detection (``this.activatedRoute.queryParams.subscribe(...)`` or maybe just the ``setSsoMode()`` function including the ``this.router.navigate(...)`` into the ``AuthenticationService``?

Reply:
Oh yeah, that's a better separation of concerns.